### ISSUE TITLE
Review: First Pass Modeling

### ISSUE THREAD

**Issue by samsoe** [mosscoder/mpg-horses#29]
- Initial development is in src/
  - Cursor has trouble editing notebooks. This slows down the exploratory dev process. Hence, the initial development is in src/ with .py scripts.
- First step was to set up a basic PyTorch framework with the Iris dataset. 
- Transitioned to the Hugging Face Horse Presence dataset.
  - Initially, the model was overfitting with a seemingly overly complex PyTorch framework
![Image](https://github.com/user-attachments/assets/375160b6-e5db-4b87-82a0-370780a6ecfc)

  - Adjusting for overfitting triggered the patience variable at a less than desirable result
![Image](https://github.com/user-attachments/assets/24521267-2968-4e4a-9878-98da71601879)

- Next steps
  - Review the encoding of the images in the output .parquet files
  - Continue with hyperparameter tuning
  - More epochs
    - Training up to this point has been with Metal on Apple M3
    - Train with CUDA

**Comment by mosscoder** [comment]
For this first past, I'd like everything to be self-contained in a single notebook. The first effort must not be thousands of lines of code spread across multiple scripts, but a clear demonstration that you understand the fundamentals. Here's a brief checklist to start you off

- Loading hugging face data, showing a batch from the dataloader
- Did you apply normalization to the images? This is critical 
- Clarify what data transformations you apply in the dataloader, such as flips, rotations, crops
- Resnet50 is a good start for architecture, and the only changes you make to this model is to attach a new model head
- Use default settings for the optimizer, and tune only learning rate and epoch count to start
- Don't include learning rate schedulers at this point

**Comment by samsoe** [comment]
I agree. I learned a lot working with the Agent, getting things going and seeing some iterative approaches. Claude went too hard too fast.
